org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLLanguageServer.java

```diff
@@ -304,7 +305,7 @@ public DOMDocument getDocument(String uri) {
 	@Override
 	public void sendNotification(String message, MessageType messageType, Command... commands) {
 		SharedSettings sharedSettings = getSharedSettings();
-		if (sharedSettings.isActionableNotificationSupport() && sharedSettings.isOpenSettingsCommandSupport()) {
+		if (sharedSettings.isActionableNotificationSupport() && sharedSettings.getCommandCapabilities().isCommandSupported(ClientCommands.OPEN_SETTINGS)) {

```

Comment:
In shared settings, I will define `isCommandSupported` (which will call getCommandCapabilities().isCommandSupported), to write only:

`sharedSettings..isCommandSupported(ClientCommands.OPEN_SETTINGS)`